Modelica/Blocks/Tables.mo

```diff
@@ -1310,35 +1262,34 @@ that can be used as a template for own developments. While &quot;usertab.c&quot;
 &quot;usertab.h&quot; needs to adapted for the own needs.</p>
 <p>In order to work it is necessary that the compiler pulls in the &quot;usertab.c&quot; file. Different Modelica tools might provide different mechanisms to do so.
 Please consult the respective documentation/support for your Modelica tool.</p>
-<p>A possible (though a bit &quot;hackish&quot;) Modelica standard conformant approach is to pull in the required files by utilizing a &quot;dummy&quot;-function that uses the Modelica external function
-interface to pull in the required &quot;usertab.c&quot;. An example how this can be done is given below.</p>
+<p>A possible (though slightly makeshift) approach is to pull in the required files by utilizing a &quot;dummy&quot;-function that uses the Modelica external function
+interface to include the required &quot;usertab.c&quot;. An example how this can be done is given below.</p>
 <blockquote><pre>
-model Test25_usertab \"Test utilizing the usertab.c interface\"
+model ExampleCTable \"Example utilizing the usertab.c interface\"
   extends Modelica.Icons.Example;
-public
-  Modelica.Blocks.Sources.RealExpression realExpression(y=getUsertab(t_new.y))
-    annotation (Placement(transformation(extent={{-40,-34},{-10,-14}})));
-  Modelica.Blocks.Tables.CombiTable1Dv t_new(tableOnFile=true, tableName=\"TestTable_1D_a\")
+  parameter Real dummy(fixed=false) \"Dummy parameter\";
+  Modelica.Blocks.Tables.CombiTable1Dv table(tableOnFile=true, tableName=\"TestTable_1D_a\")
     annotation (Placement(transformation(extent={{-40,0},{-20,20}})));
   Modelica.Blocks.Sources.ContinuousClock clock
     annotation (Placement(transformation(extent={{-80,0},{-60,20}})));
 protected
-  encapsulated function getUsertab
+  encapsulated impure function getUsertab \"External dummy function to include \\\"usertab.c\\\"\"
```

Comment:
Why is usertab marked impure here? It is pure in this case, and if the function is not pure... does the rest of the tables work as it should?

Reply:
Because it must not be removed during optimization. It is a makeshift solution to force the external Include annotation to get effective.

Reply:
This was one of the main questions of this PR. See e.g. @HansOlsson's https://github.com/modelica/ModelicaStandardLibrary/pull/2858#issuecomment-496432733.